src/main/kotlin/de/itemis/mps/gradle/generate/Tasks.kt

```diff
@@ -0,0 +1,39 @@
+package de.itemis.mps.gradle.generate
+
+import org.gradle.api.DefaultTask
+import org.gradle.api.GradleException
+import org.gradle.api.tasks.InputDirectory
+import org.gradle.api.tasks.InputFile
+import org.gradle.api.tasks.OutputFile
+import org.gradle.api.tasks.TaskAction
+import java.io.File
+import java.io.PrintWriter
+import java.util.*
+
+
+open class FakeBuildNumberTask: DefaultTask() {
```

Comment:
What is this used for?

Reply:
The problem is that for some reason the `ApplicationInfo` isn't initialised properly. That causes `PluginManagerCore.BUILD_NUMBER` to be null. 

In this case the `PluginManagerCore` resorts to `BuildNumber.currentVersion()` which finally calls into `BuildNumber.fromFile()`. That allows us to place a `build.txt` in the root of the home path (see `PathManager.getHomePath()`). Which is then used to load the build number from. 

I just saw that at least for 2018.2 a newer version of the platform is shipped which allows to get a similar behaviour via setting `idea.plugins.compatible.build` property.

Reply:
Can you add this information as a doc comment to the class?

Reply:
Added it to the `Plugin.kt` file where the task is used.